modules/axtask/src/run_queue.rs

```diff
@@ -1,72 +1,249 @@
+///
+/// A static reference to the current run queue.
+// #[inline(always)]
+pub(crate) fn current_run_queue<G: BaseGuard>() -> AxRunQueueRef<'static, G> {
+    let irq_state = G::acquire();
+    AxRunQueueRef {
+        inner: unsafe { RUN_QUEUE.current_ref_mut_raw() },
+        state: irq_state,
+        _phantom: core::marker::PhantomData,
+    }
+}
 
-static WAIT_FOR_EXIT: WaitQueue = WaitQueue::new();
+/// Selects the run queue index based on a CPU set bitmap, minimizing the number of tasks.
+///
+/// This function filters the available run queues based on the provided `cpumask` and
+/// selects the one with the fewest tasks. The selected run queue's index (cpu_id) is returned.
+///
+/// ## Arguments
+///
+/// * `cpumask` - A bitmap representing the CPUs that are eligible for task execution.
+///
+/// ## Returns
+///
+/// The index (cpu_id) of the selected run queue.
+///
+/// ## Panics
+///
+/// This function will panic if there is no available run queue that matches the CPU set.
+///
+#[cfg(feature = "smp")]
+#[inline]
+fn select_run_queue_index(cpumask: CpuMask<{ axconfig::SMP }>) -> usize {
+    unsafe {
+        RUN_QUEUES
+            .iter()
+            .filter(|rq| cpumask.get(rq.assume_init_ref().cpu_id()))
+            .min_by_key(|rq| rq.assume_init_ref().num_tasks())
+            .expect("No available run queue that matches the CPU set")
+            .assume_init_ref()
+            .cpu_id()
+    }
+}
 
-#[percpu::def_percpu]
-static IDLE_TASK: LazyInit<AxTaskRef> = LazyInit::new();
+/// Retrieves a `'static` reference to the run queue corresponding to the given index.
+///
+/// This function asserts that the provided index is within the range of available CPUs
+/// and returns a reference to the corresponding run queue.
+///
+/// ## Arguments
+///
+/// * `index` - The index of the run queue to retrieve.
+///
+/// ## Returns
+///
+/// A reference to the `AxRunQueue` corresponding to the provided index.
+///
+/// ## Panics
+///
+/// This function will panic if the index is out of bounds.
+///
+#[inline]
+fn get_run_queue(index: usize) -> &'static mut AxRunQueue {
+    assert!(index < axconfig::SMP);
+    unsafe { RUN_QUEUES[index].assume_init_mut() }
+}
 
+/// Selects the appropriate run queue for the provided task.
+///
+/// * In a single-core system, this function always returns a reference to the global run queue.
+/// * In a multi-core system, this function selects the run queue based on the task's CPU affinity and load balance.
+///
+/// ## Arguments
+///
+/// * `task` - A reference to the task for which a run queue is being selected.
+///
+/// ## Returns
+///
+/// A reference to the selected `AxRunQueue`.
+///
+/// ## TODO
+///
+/// 1. Implement better load balancing across CPUs for more efficient task distribution.
+/// 2. Use a more generic load balancing algorithm that can be customized or replaced.
+///
+#[inline]
+pub(crate) fn select_run_queue<G: BaseGuard>(
+    #[cfg(feature = "smp")] task: AxTaskRef,
```

Comment:
Please avoid using conditional compilation in function arguments.

Reply:
ok, it'll be fixed soon